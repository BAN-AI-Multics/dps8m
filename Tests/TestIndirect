de DSBR.ADDR 01000
de dsbr.u 1
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM

load Play/TestIndirect.oct segment 0 address 0
;load Play/PlayA.oct deferred
load Play/UtilsA.oct deferred
load Play/simh_operators.oct deferred
;load Play/power_of_two.oct deferred
load Play/TestIndirectA.oct deferred

segments lot create
segments create stack 0
segments resolve
segments load deferred
segments lot snap

;set debug stderr
;set cpu debug=trace;regdumpaqi;regdumpidx;addrmod;appending;REGDUMPPR
;set cpu debug=trace;addrmod;appending

g 0
q

