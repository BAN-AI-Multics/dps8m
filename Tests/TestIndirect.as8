
    option  Multics     " Multics style calling conventions

    entrypoint main
    
	name	main
	segdef	main

	segref	UtilsA, ioa,rd_clockA
	segref	lot_,lot_
    segref  stack_0, stack_0

    segref  TAM, _main, pi, E, fltTmp, _print

    " our indirection tests ....
    segref  TIND, _main_2
    
    " our FXE indirection tests ...
    segref  TINDC, _main_C

    "segref  indirect_tests

    link    I, <UtilsA>|[ioa]
    link    D, <UtilsA>|[rd_clockA]

    link    T, <TAM>|[_main]
    link    P, <TAM>|[_print]

main:
    tsx2    setupFXE
    "push

"
" put a test for abs mode ind addr here ...
"



    LDA     I
    tsp2	pr4|I, *        " ioa
    acc     /Howdy world this is 'TestIndirect'! - It's /

    tsp2    rd_clockA
    "tsp2    <UtilsA>|[rd_clockA]
    "tsp2    lp|D,*
    "tsp2    UtilsA$rd_clockA

    staq	Date        " MM/DD HH:MM
        
    tsp2	ioa
    acc     /'^g^g'\r\n/
    nop     Date
    nop     Date+1

   " sprpap      g
   " sprp4       f
   " short_call  TAM_main
   " lprp4       f
   " lprpap      g


    "call    P(ap|0)

    "call    TAM_main(ap|0)
    call T(ap|0)

    "tra     mmm
    
    "
    " dink with a few fp numbers in TAM$
    "
    fld     pi
    emcall  6

    fld     E
    emcall  6

    fmp     pi  " pi * e = 8.5397341
    emcall  6

    call    P(ap|0)     " TAM print (again)
    
mmm:

    "
    " Now lets Setup with some indirection tests for an indirection chain ...
    "
TAMmA:
    epp0    TAM$magicA
    spri0    magicM
    ldq     =o20        " o20 => Q (set indirection mod for ITS pointer)
    stcq    magicM+1, 5

    tsp2    ioa
    acc     /TestIndirect: magicM: ^w ^w\r\n\r\n/
    nop     magicM+0
    nop     magicM+1

    link    TIND_main, <TIND>|[_main_2]
    call    TIND_main (ap|0)

    "
    " for C ...
    "
    link    _mainC, <TINDC>|[_main_C]
    call    _mainC (ap|0)


allDone:
    tsp2	ioa
    acc     /T-H-A-T-S all folks (from 0|main) !!! /

	dis
    

	even

g:  bss 1
f:  bss 1
LA: bss 1
LQ: bss 1

Date:	bss     2

    segdef  Data_1

Data_1: oct 12345654321

    segref  TAM, magicA
    segdef  magicM
    even
magicM: zero    0,0 "arg TAM$magicA,*



    INCLUDE     setupFXE.as8

	end



