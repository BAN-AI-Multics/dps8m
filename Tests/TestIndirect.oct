!SIZE 000200
!SEGNAME main
!SEGDEF main 000000
!LINKAGE 000142 15
!SEGREF UtilsA ioa 000142
!SEGREF UtilsA rd_clockA 000144
!SEGREF lot_ lot_ 000146
!SEGREF stack_0 stack_0 000150
!SEGREF TAM _main 000152
!SEGREF TAM pi 000154
!SEGREF TAM E 000156
!SEGREF TAM fltTmp 000160
!SEGREF TAM _print 000162
!SEGREF simh_operators_ operator_table 000164
!SEGREF simh_operators_ operator_table 000166 +632
!SEGREF simh_operators_ operator_table 000170 +633
!SEGREF simh_operators_ operator_table 000172 +634
!SEGREF simh_operators_ operator_table 000174 +635
!SEGREF simh_operators_ operator_table 000176 +636
!GO 000000
000000 xxxx 000070702000     tsx2    setupFXE
000001 xxxx 000060627000     push
000002 xxxx 700040272120 
000003 xxxx 000000235000     LDA     I
000004 xxxx 400000272120     tsp2	pr4|I, *        " ioa
000005 xxxx 053110157167     acc     /Howdy world this is 'TestIndirect'! - It's /
000006 xxxx 144171040167 
000007 xxxx 157162154144 
000010 xxxx 040164150151 
000011 xxxx 163040151163 
000012 xxxx 040047124145 
000013 xxxx 163164111156 
000014 xxxx 144151162145 
000015 xxxx 143164047041 
000016 xxxx 040055040111 
000017 xxxx 164047163040 
000020 xxxx 400002272120     tsp2    rd_clockA
000021 xxxx 000066757000     staq	Date        " MM/DD HH:MM
000022 xxxx 400000272120     tsp2	ioa
000023 xxxx 010047136147     acc     /'^g^g'\r\n/
000024 xxxx 136147047015 
000025 xxxx 012040040040 
000026 xxxx 000066011000     nop     Date
000027 xxxx 000067011000     nop     Date+1
000030 xxxx 600000254100     call    P(ap|0)
000031 xxxx 600040753100 
000032 xxxx 000000350100 
000033 xxxx 400020352120 
000034 xxxx 700036670120 
000035 xxxx 600000173100 
000036 xxxx 600040073100 
000037 xxxx 600000254100     call T(ap|0)
000040 xxxx 600040753100 
000041 xxxx 000000350100 
000042 xxxx 400010352120 
000043 xxxx 700036670120 
000044 xxxx 600000173100 
000045 xxxx 600040073100 
000046 xxxx 400012431120     fld     pi
000047 xxxx 000006420400     emcall  6
000050 xxxx 400014431120     fld     E
000051 xxxx 000006420400     emcall  6
000052 xxxx 400012461120     fmp     pi  " pi * e = 8.5397341
000053 xxxx 000006420400     emcall  6
000054 xxxx 600000254100     call    P(ap|0)
000055 xxxx 600040753100 
000056 xxxx 000000350100 
000057 xxxx 400020352120 
000060 xxxx 700036670120 
000061 xxxx 600000173100 
000062 xxxx 600040073100 
000063 xxxx 000000616000 	dis
000064 xxxx 000000000000 g:  bss 1
000065 xxxx 000000000000 f:  bss 1
000066 xxxx 000000000000 Date:	bss     2
000070 xxxx 400004764120         lprplp  lot_            " set-up lp to point to this segments linkage section
000071 xxxx 400006372120         eppsp   stack_0         " set-up stack pointer
000072 xxxx 400006373520         eppsb   stack_0         " set-up stack base
000073 xxxx 400006373520     eppsb   stack_0
000074 xxxx 400006372120     eppsp   stack_0
000075 xxxx 000100152003     adwpsp  =0100,du
000076 xxxx 700022652100     sprisp	sb|stack_header.stack_begin_ptr 		" save beginning of 1st stack frame
000077 xxxx 700024652100     sprisp	sb|stack_header.stack_end_ptr
000100 xxxx 400004352120 	eppbp	lot_						" set the lot ptrs
000101 xxxx 700026252100 	spribp	sb|stack_header.lot_ptr
000102 xxxx 700052252100 	spribp	sb|stack_header.isot_ptr
000103 xxxx 000136352020 	eppbp	=its(-2,2),*		" signal ptr terminates process
000104 xxxx 700030252100 	spribp	sb|stack_header.signal_ptr
000105 xxxx 400022352120 	eppbp	simh_operators_$operator_table  		" Get a ptr to  PL/I operators table.
000106 xxxx 700034252100 	spribp	sb|stack_header.pl1_operators_ptr 	" Store in  inzr_stk0
000107 xxxx 400024352120 	eppbp	simh_operators_$operator_table+call_offset,*
000110 xxxx 700036252100 	spribp	sb|stack_header.call_op_ptr
000111 xxxx 400026352120 	eppbp	simh_operators_$operator_table+push_offset,*
000112 xxxx 700040252100 	spribp	sb|stack_header.push_op_ptr
000113 xxxx 400030352120 	eppbp	simh_operators_$operator_table+return_offset,*
000114 xxxx 700042252100 	spribp	sb|stack_header.return_op_ptr
000115 xxxx 400032352120 	eppbp	simh_operators_$operator_table+return_no_pop_offset,*
000116 xxxx 700044252100 	spribp	sb|stack_header.ret_no_pop_op_ptr
000117 xxxx 400034352120 	eppbp	simh_operators_$operator_table+entry_offset,*
000120 xxxx 700046252100 	spribp	sb|stack_header.entry_op_ptr
000121 xxxx 000140352020 	eppbp	=its(-1,1),*
000122 xxxx 700070252100 	spribp	sb|stack_header.heap_header_ptr
000123 xxxx 700046272120     tsp2 pr7|stack_header.entry_op_ptr,*
000124 xxxx 000060627000     push
000125 xxxx 700040272120 
000126 xxxx 000000710012 	tra		0,2				" ... and return to caller
000127 xxxx 000000011000 	even "(allocating 1 nop)
000130 xxxx 000000000004 no_args:	zero	0,4
000131 xxxx 000000000000 	zero	0,0
000132 xxxx 000000000000 decimal literal (=0)
000133 xxxx 000000000000 decimal literal (=0)
000134 xxxx 000000000000 decimal literal (=0)
000136 xxxx 077776000043 its literal (=its(77776,000002,00))
000137 xxxx 000002000000 
000140 xxxx 077777000043 its literal (=its(77777,000001,00))
000141 xxxx 000001000000 
000142 xxxx 000000000043 link UtilsA$ioa
000143 xxxx 000000000000 
000144 xxxx 000000000043 link UtilsA$rd_clockA
000145 xxxx 000000000000 
000146 xxxx 000000000043 link lot_$lot_
000147 xxxx 000000000000 
000150 xxxx 000000000043 link stack_0$stack_0
000151 xxxx 000000000000 
000152 xxxx 000000000043 link TAM$_main
000153 xxxx 000000000000 
000154 xxxx 000000000043 link TAM$pi
000155 xxxx 000000000000 
000156 xxxx 000000000043 link TAM$E
000157 xxxx 000000000000 
000160 xxxx 000000000043 link TAM$fltTmp
000161 xxxx 000000000000 
000162 xxxx 000000000043 link TAM$_print
000163 xxxx 000000000000 
000164 xxxx 000000000043 link simh_operators_$operator_table
000165 xxxx 000000000000 
000166 xxxx 000000000043 link simh_operators_$operator_table+632
000167 xxxx 001170000000 
000170 xxxx 000000000043 link simh_operators_$operator_table+633
000171 xxxx 001171000000 
000172 xxxx 000000000043 link simh_operators_$operator_table+634
000173 xxxx 001172000000 
000174 xxxx 000000000043 link simh_operators_$operator_table+635
000175 xxxx 001173000000 
000176 xxxx 000000000043 link simh_operators_$operator_table+636
000177 xxxx 001174000000 
