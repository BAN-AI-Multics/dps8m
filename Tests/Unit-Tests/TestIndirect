de DSBR.ADDR 01000
de dsbr.u 1
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM

load TestIndirect.oct segment 0 address 0
load TestIndirectA.oct deferred
load TestIndirectB.oct deferred
load TestIndirectC.oct deferred

load UtilsA.oct deferred
load simh_operators.oct deferred

segments lot create
segments create stack 0
segments resolve
segments load deferred
segments lot snap

;set debug stderr
;set cpu debug=trace;regdumpaqi;regdumpidx;addrmod;appending;REGDUMPPR
;set cpu debug=trace;addrmod;appending;REGDUMPPR

g 1 
q

