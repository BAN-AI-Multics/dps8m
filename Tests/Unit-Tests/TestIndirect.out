zero-initialized segments 0 .. 7
zero-initialized SDWAM
segment created for 'main'
segdef created for segment main, symbol 'main', addr:000001
segdef created for segment main, symbol 'Data_1', addr:000154
segdef created for segment main, symbol 'magicM', addr:000156
segref created for segment 'UtilsA' symbol:ioa, addr:000232
segref created for segment 'UtilsA' symbol:rd_clockA, addr:000234
segref created for segment 'lot_' symbol:lot_, addr:000236
segref created for segment 'stack_0' symbol:stack_0, addr:000240
segref created for segment 'TAM' symbol:_main, addr:000242
segref created for segment 'TAM' symbol:pi, addr:000244
segref created for segment 'TAM' symbol:E, addr:000246
segref created for segment 'TAM' symbol:fltTmp, addr:000250
segref created for segment 'TAM' symbol:_print, addr:000252
segref created for segment 'TIND' symbol:_main_2, addr:000254
segref created for segment 'TINDC' symbol:_main_C, addr:000256
segref created for segment 'TINDC' symbol:magicC, addr:000260
segref created for segment 'TAM' symbol:magicA, addr:000262
segref created for segment 'TAM' symbol:magicA, addr:000264
segref created for segment 'simh_operators_' symbol:operator_table, addr:000266
segref created for segment 'simh_operators_' symbol:operator_table+632, addr:000270
segref created for segment 'simh_operators_' symbol:operator_table+633, addr:000272
segref created for segment 'simh_operators_' symbol:operator_table+634, addr:000274
segref created for segment 'simh_operators_' symbol:operator_table+635, addr:000276
segref created for segment 'simh_operators_' symbol:operator_table+636, addr:000300
!GO address: 000001
Writing SDW to address 00001000 (DSBR.ADDR+2*0 offset) 
193 (000301) words loaded into segment 0(0) at address 000000
segment created for 'TAM'
segdef created for segment TAM, symbol 'pi', addr:001723
segdef created for segment TAM, symbol 'E', addr:001724
segdef created for segment TAM, symbol 'fltTmp', addr:001725
segdef created for segment TAM, symbol 'magicA', addr:000004
entrypoint created for segment main, symbol '_main', addr:001752
entrypoint created for segment main, symbol '_print', addr:001754
segref created for segment 'UtilsA' symbol:ioa, addr:001756
segref created for segment 'TIND' symbol:magicB, addr:001760
996 (001744) words loaded into segment TAM
segment created for 'TIND'
segdef created for segment TIND, symbol 'magicB', addr:002324
entrypoint created for segment main, symbol '_main_2', addr:002562
segref created for segment 'UtilsA' symbol:ioa, addr:002570
segref created for segment 'main' symbol:Data_1, addr:002572
segref created for segment 'main' symbol:magicM, addr:002574
151 (000227) words loaded into segment TIND
segment created for 'TINDC'
segdef created for segment TINDC, symbol 'magicC', addr:000025
entrypoint created for segment main, symbol '_main_C', addr:000410
segref created for segment 'UtilsA' symbol:ioa, addr:000422
251 (000373) words loaded into segment TINDC
segment created for 'UtilsA'
segdef created for segment UtilsA, symbol 'A0', addr:000005
segdef created for segment UtilsA, symbol 'A1', addr:000050
segdef created for segment UtilsA, symbol 'rd_clockA', addr:000207
segdef created for segment UtilsA, symbol 'ioa', addr:000271
857 (001531) words loaded into segment UtilsA
segment created for 'simh_operators_'
segdef created for segment simh_operators_, symbol 'alm_call', addr:001175
segdef created for segment simh_operators_, symbol 'alm_entry', addr:001215
segdef created for segment simh_operators_, symbol 'alm_operators_begin', addr:001175
segdef created for segment simh_operators_, symbol 'alm_operators_end', addr:001237
segdef created for segment simh_operators_, symbol 'alm_push', addr:001201
segdef created for segment simh_operators_, symbol 'alm_return', addr:001225
segdef created for segment simh_operators_, symbol 'alm_return_no_pop', addr:001233
segdef created for segment simh_operators_, symbol 'transfer_vector', addr:000551
segdef created for segment simh_operators_, symbol 'operator_table', addr:000000
40 (000050) words loaded into segment simh_operators_
lot_ segment created with 6 sparse entries.
stack_0 segment created as segment# 8 (10) [DSBR.STACK=0001]
Examining segments ... segment numbering begins at: 9
    Processing segment main...
        Resolving segref UtilsA$ioa...found ioa (000232)
            ITS Pair: [even:000011000043, odd:000271000000]
        Resolving segref UtilsA$rd_clockA...found rd_clockA (000234)
            ITS Pair: [even:000011000043, odd:000207000000]
        Resolving segref lot_$lot_...found lot_ (000236)
            ITS Pair: [even:000012000043, odd:000000000000]
        Resolving segref stack_0$stack_0...found stack_0 (000240)
            ITS Pair: [even:000010000043, odd:000000000000]
        Resolving segref TAM$_main...found _main (000242)
            ITS Pair: [even:000013000043, odd:001752000000]
        Resolving segref TAM$pi...found pi (000244)
            ITS Pair: [even:000013000043, odd:001723000000]
        Resolving segref TAM$E...found E (000246)
            ITS Pair: [even:000013000043, odd:001724000000]
        Resolving segref TAM$fltTmp...found fltTmp (000250)
            ITS Pair: [even:000013000043, odd:001725000000]
        Resolving segref TAM$_print...found _print (000252)
            ITS Pair: [even:000013000043, odd:001754000000]
        Resolving segref TIND$_main_2...found _main_2 (000254)
            ITS Pair: [even:000014000043, odd:002562000000]
        Resolving segref TINDC$_main_C...found _main_C (000256)
            ITS Pair: [even:000015000043, odd:000410000000]
        Resolving segref TINDC$magicC...found magicC (000260)
            ITS Pair: [even:000015000043, odd:000025000000]
        Resolving segref TAM$magicA...found magicA (000262)
            ITS Pair: [even:000013000043, odd:000004000000]
        Resolving segref TAM$magicA...found magicA (000264)
            ITS Pair: [even:000013000043, odd:000004000000]
        Resolving segref simh_operators_$operator_table...found operator_table (000266)
            ITS Pair: [even:000016000043, odd:000000000000]
        Resolving segref simh_operators_$operator_table...found operator_table+632 (000270)
            ITS Pair: [even:000016000043, odd:001170000000]
        Resolving segref simh_operators_$operator_table...found operator_table+633 (000272)
            ITS Pair: [even:000016000043, odd:001171000000]
        Resolving segref simh_operators_$operator_table...found operator_table+634 (000274)
            ITS Pair: [even:000016000043, odd:001172000000]
        Resolving segref simh_operators_$operator_table...found operator_table+635 (000276)
            ITS Pair: [even:000016000043, odd:001173000000]
        Resolving segref simh_operators_$operator_table...found operator_table+636 (000300)
            ITS Pair: [even:000016000043, odd:001174000000]
    Processing segment TAM...
        Resolving segref UtilsA$ioa...found ioa (001756)
            ITS Pair: [even:000011000043, odd:000271000000]
        Resolving segref TIND$magicB...found magicB (001760)
            ITS Pair: [even:000014000043, odd:002324000000]
    Processing segment TIND...
        Resolving segref UtilsA$ioa...found ioa (002570)
            ITS Pair: [even:000011000043, odd:000271000000]
        Resolving segref main$Data_1...found Data_1 (002572)
            ITS Pair: [even:000000000043, odd:000154000000]
        Resolving segref main$magicM...found magicM (002574)
            ITS Pair: [even:000000000043, odd:000156000000]
    Processing segment TINDC...
        Resolving segref UtilsA$ioa...found ioa (000422)
            ITS Pair: [even:000011000043, odd:000271000000]
    Processing segment UtilsA...
    Processing segment simh_operators_...
    Processing segment lot_...
    Processing segment stack_0...
Loading deferred segments ...
    loading TAM as segment# 013
Writing SDW to address 00001026 (DSBR.ADDR+2*11 offset) 
      1010 (001762) words loaded into segment 11 (13) at address 201000
    loading TIND as segment# 014
Writing SDW to address 00001030 (DSBR.ADDR+2*12 offset) 
      1406 (002576) words loaded into segment 12 (14) at address 203000
    loading TINDC as segment# 015
Writing SDW to address 00001032 (DSBR.ADDR+2*13 offset) 
      276 (000424) words loaded into segment 13 (15) at address 205600
    loading UtilsA as segment# 011
Writing SDW to address 00001022 (DSBR.ADDR+2*9 offset) 
      1064 (002050) words loaded into segment 9 (11) at address 206240
    loading simh_operators_ as segment# 016
Writing SDW to address 00001034 (DSBR.ADDR+2*14 offset) 
      671 (001237) words loaded into segment 14 (16) at address 210320
    loading lot_ as segment# 012
Writing SDW to address 00001024 (DSBR.ADDR+2*10 offset) 
      4095 (007777) words loaded into segment 10 (12) at address 211560
LOT => PR[4]: SNR=00012 RNR=0 WORDNO=000000 BITNO:00
LOT => PR[5]: SNR=00012 RNR=0 WORDNO=000000 BITNO:00
    loading stack_0 as segment# 010
Writing SDW to address 00001020 (DSBR.ADDR+2*8 offset) 
      49152 (140000) words loaded into segment 8 (10) at address 221560
snapping lot_ links...

dps-8/m simulator V4.0-0 Beta        git commit id: c8ae76cb
	211560 + 0 => 000000000232
	211560 + 13 => 000013001756
	211560 + 14 => 000014002570
	211560 + 15 => 000015000422
Howdy world this is 'TestIndirect'! - It's ' 6/19 10:06 '
TINDC$magicC was = 2718281828 (Should be 2718281828)
TestIndirectA: magicA: 14 000043 2324 000000

R mods ...
'If this works, that means were in TAM$...'
Should be ... 006000 0100 03 is 6000010003
Should be ... 006000 0100 03 is 6000010003
Should be ... 400000000003 is 400000000003
Should be ... 000000000012 is 12
Should be ... 000005 000004 is 000005 000004
Should be ... 5 is 5
Should be ... 17 is 17
Should be ... 000006777776 (6,-2) is 000006 777776
Should be ... 3 is 3
Should be ... 700 is 700
Should be ... 000000235004 is 000000 235004
Should be ... 123456000000/000000543210 is 123456000000/543210
RI mods ...
Should be ... 42/400000000002 is 42/400000000002
Should be ... 616 is 616
Should be ... 77 is 77
Should be ... 2012 is 2012
Should be ... ??? (since l+2 & m move) is 1623000000
Should be ... 400000000003 is 400000000003
Should be ... 3 is 3
Should be ... 42 is 42
IR mods ...
Should be ... 400000000001 is 400000000001
Should be ... ??? (this will move as x moves) is 000000 001647
Should be ... 377777777523 is 377777777523
Should be ... 000000000222 is 222
Should be ... 000000000333 is 333
Should be ... 00000000042 is 42
Should be ... ??? is 1643 (Yup, &w. But, &w will change)
Should be ... 616 is 616
Should be ... 00000000013 is 13
Should be ... 0000000002012 is 2012
Should be ... 34359738195 is 34359738195
Should be ... 000007777777 is 7777777
IT mods ...
Should be ... 000000000046 .. 000000000034 is 46
Should be ... 000000000046 .. 000000000034 is 44
Should be ... 000000000046 .. 000000000034 is 42
Should be ... 000000000046 .. 000000000034 is 40
Should be ... 000000000046 .. 000000000034 is 36
Should be ... 000000000046 .. 000000000034 is 34
Should be ... 000000000013 .. 000000000025 is 13
Should be ... 000000000013 .. 000000000025 is 15
Should be ... 000000000013 .. 000000000025 is 17
Should be ... 000000000013 .. 000000000025 is 21
Should be ... 000000000013 .. 000000000025 is 23
Should be ... 000000000013 .. 000000000025 is 25
Should be ... 'This ' is 'T'
Should be ... 'This ' is 'h'
Should be ... 'This ' is 'i'
Should be ... 'This ' is 's'
Should be ... 'This ' is ' '
Should be ... 'gfedc' is 'g'
Should be ... 'gfedc' is 'f'
Should be ... 'gfedc' is 'e'
Should be ... 'gfedc' is 'd'
Should be ... 'gfedc' is 'c'
Should be ... '000I' is '00000I'
Should be ... '000B' is '00000B'
Should be ... '000M' is '00000M'
Should be ... 42 is 42
Should be ... 616,42,5,4,3,2,1 is 616
Should be ... 616,42,5,4,3,2,1 is 42
Should be ... 616,42,5,4,3,2,1 is 5
Should be ... 616,42,5,4,3,2,1 is 4
Should be ... 616,42,5,4,3,2,1 is 3
Should be ... 616,42,5,4,3,2,1 is 2
Should be ... 616,42,5,4,3,2,1 is 1
Should be ... 100, 200, ..., 700, 7777 is 100
Should be ... 100, 200, ..., 700, 7777 is 200
Should be ... 100, 200, ..., 700, 7777 is 300
Should be ... 100, 200, ..., 700, 7777 is 400
Should be ... 100, 200, ..., 700, 7777 is 500
Should be ... 100, 200, ..., 700, 7777 is 600
Should be ... 100, 200, ..., 700, 7777 is 700
Should be ... 100, 200, ..., 700, 7777 is 7777
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 400000000001
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 400000000002
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 400000000003
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 0
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 1
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 2
Should be ... o-1, o-2, o-3, 0, 1, 2, 3 is 3
Should be ... 100 .. 700 is 100
Should be ... 100 .. 700 is 200
Should be ... 100 .. 700 is 300
Should be ... 100 .. 700 is 400
Should be ... 100 .. 700 is 500
Should be ... 100 .. 700 is 600
Should be ... 100 .. 700 is 700
Should be ... -1, 0, o13 is 400000000001
Should be ... -1, 0, o13 is 0
Should be ... -1, 0, o13 is 13
Should be ... 012, 345, o-255, 670 is 12
Should be ... 012, 345, o-255, 670 is 345
Should be ... 012, 345, o-255, 670 is 400000000255
Should be ... 012, 345, o-255, 670 is 670
That's all folks!!! (TAM$_main)
   3.1415927   2.7182818   8.5397341
This is from TAM$_print ...
   3.1415927   2.7182818   -123.4567
TestIndirect: magicM: 000013 000043 000004 000020

Hallooooo from TIND$_main...
main$Data_1 a: 12345654321 (Should be 12345654321)
main$Data_1 b: 666 (Should be 666)
main$Data_1 c: 49510627 (Should be 49510627 [Thanks Lexi!])
main$Data_1 d: 982187 (Should be 982187 [Thanks Lydia!])
main$magicM 1: 3141592653 (Should be 3141592653)
main$magicM 2: 412345654321 (Should be 412345654321)
Now leaving TIND$_main ...
This is TINC$_main_C...
_main_C:1: magicC = 31415926536 (Should be 31415926536)
_main_C:1: t1 2718271828 (addr:Even:000010 000043 Odd:000230 000000)
_main_C:2: d1 236776060576 265223074736 (addr:Even:000010 000043 Odd:000246 000000)
_main_C:3: addr t2 Even:000010 000043 Odd:000231 000000)
_main_C:4: filling t2 with 13 values from 30..42 : 30 31 32 33 34 35 36 37 38 39 40 41 42 
_main_C:5: addr d2 Even:000010 000043 Odd:000250 000000)
   8.5397342
   26.828366
   84.283798
   264.78536
   831.84775
   2613.3268
   8210.0082
   25792.501
   81029.533
   254561.79
   799729.43
   2512424.1
   7893013.1
    24796632
    77900917
2.4473295e+08

_printTemps:1: t1 2718271828 (Should be: 2718271828)
_printTemps:2: d1 236776060576 265223074736 (Should be: 236776060576 265223074736)
_printTemps:3: t2(0)=30
_printTemps:3: t2(1)=31
_printTemps:3: t2(2)=32
_printTemps:3: t2(3)=33
_printTemps:3: t2(4)=34
_printTemps:3: t2(5)=35
_printTemps:3: t2(6)=36
_printTemps:3: t2(7)=37
_printTemps:3: t2(8)=38
_printTemps:3: t2(9)=39
_printTemps:3: t2(10)=40
_printTemps:3: t2(11)=41
_printTemps:3: t2(12)=42
Now leaving TINC$_main_C ...
T-H-A-T-S all folks (from 0|main) !!! 

cpuCycles = 118430
TestIndirect-29> g 1 

DIS instruction, IC: main|145 (00000145) (000000616000)
Goodbye
